app/services/decidim/action_delegator/sms_gateway.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module Decidim
+  module ActionDelegator
+    class SmsGateway
+      attr_reader :mobile_phone_number, :code, :response
+
+      def initialize(mobile_phone_number, code)
+        @mobile_phone_number = mobile_phone_number
+        @code = code
+      end
+
+      def deliver_code
+        send_sms!
+
+        success?
+      end
+
+      private
+
+      def success?
+        parsed_response[:code] == "200"
+      end
+
+      def parsed_response
+        return @parsed_response if @parsed_response
+
+        doc = Nokogiri::XML response.body[:send_sms_response][:result]
+
+        @parsed_response = {
+          code: doc.xpath("//codigo").text,
+          description: doc.xpath("//descripcion").text
+        }
+
+        @parsed_response
+      end
+
+      def send_sms!
+        @response = client.call(:send_sms,
+                                message: {
+                                  user: ENV["SMS_USER"],
+                                  pass: ENV["SMS_PASS"],
+                                  src: sender_name,
+                                  dst: mobile_phone_number,
+                                  msg: message
+                                })
+
+        Rails.logger.debug "==========="
+        Rails.logger.debug ["action_delegator_gateway", "send_sms!", response].map(&:inspect).join("\n")
```

Comment:
I'm not sure this will look ok in production logs given we are using Lograge with its JSON formatter. Was it only intended for development? I think it should be a hash instead if we want it for production. IMO I would do so as it's the only way to see the gateway's response.

Reply:
Oh I copied it from `permissions`, but I just realised that it's only enabled in development in there. I wanted to keep log of the api response but for sure there are better ways to do that :D I wanted to keep track of the sms that we send and also the errors. Ideas?

Reply:
I would use the info log level so we see these logs in staging and production without changing the app to debug, and simply pass a hash to the logger method with whatever we want to log.